src/backend/distributed/commands/trigger.c

```diff
@@ -203,45 +242,480 @@ get_relation_trigger_oid_compat(HeapTuple heapTuple)
+	}
+
+	EnsureCoordinator();
+}
+
+
+/*
+ * GetDropTriggerStmtRelation takes a DropStmt for a trigger
+ * object and returns RangeVar for the relation that owns the trigger.
+ */
+static RangeVar *
+GetDropTriggerStmtRelation(DropStmt *dropTriggerStmt)
+{
+	Assert(dropTriggerStmt->removeType == OBJECT_TRIGGER);
+
+	List *targetObjectList = dropTriggerStmt->objects;
+
+	/* drop trigger commands can only drop one at a time */
+	Assert(list_length(targetObjectList) == 1);
+	List *triggerObjectNameList = linitial(targetObjectList);
+
+	List *relationNameList = NIL;
+	SplitLastPointerElement(triggerObjectNameList, &relationNameList, NULL);
+
+	return makeRangeVarFromNameList(relationNameList);
+}
+
+
+/*
+ * PostprocessCreateTriggerStmt is called after a CREATE TRIGGER command has
+ * been executed by standard process utility. This function errors out for
+ * unsupported commands or creates ddl job for supported CREATE TRIGGER commands.
+ */
+List *
+PostprocessCreateTriggerStmt(Node *node, const char *commandString)
+{
+	CreateTrigStmt *createTriggerStmt = castNode(CreateTrigStmt, node);
+
+	RangeVar *relation = createTriggerStmt->relation;
+
+	bool missingOk = false;
+	LOCKMODE lockMode = ShareRowExclusiveLock;
+	Oid relationId = RangeVarGetRelid(relation, lockMode, missingOk);
+
+	if (!IsCitusTable(relationId))
+	{
+		return NIL;
+	}
+
+	ErrorIfUnsupportedTriggerCommand((Node *) createTriggerStmt, lockMode);
+
+	if (IsCreateCitusTruncateTriggerStmt(createTriggerStmt))
+	{
+		return NIL;
+	}
+
+	if (IsCitusLocalTable(relationId))
+	{
+		char *triggerName = createTriggerStmt->trigname;
+		return CitusLocalTableTriggerCommandDDLJob(relationId, triggerName,
+												   commandString);
+	}
+
+	return NIL;
+}
+
+
+/*
+ * IsCreateCitusTruncateTriggerStmt returns true if given createTriggerStmt
+ * creates citus_truncate_trigger.
+ */
+static bool
+IsCreateCitusTruncateTriggerStmt(CreateTrigStmt *createTriggerStmt)
+{
+	List *functionNameList = createTriggerStmt->funcname;
+	RangeVar *functionRangeVar = makeRangeVarFromNameList(functionNameList);
+	char *functionName = functionRangeVar->relname;
 	if (strncmp(functionName, CITUS_TRUNCATE_TRIGGER_NAME, NAMEDATALEN) == 0)
 	{
-		return;
+		return true;
 	}
 
-	char *relationName = triggerRelation->relname;
+	return false;
+}
+
+
+/*
+ * CreateTriggerEventExtendNames extends relation name and trigger name name
+ * with shardId, and sets schema name in given CreateTrigStmt.
+ */
+void
+CreateTriggerEventExtendNames(CreateTrigStmt *createTriggerStmt, char *schemaName,
+							  uint64 shardId)
+{
+	RangeVar *relation = createTriggerStmt->relation;
+
+	char **relationName = &(relation->relname);
+	AppendShardIdToName(relationName, shardId);
+
+	char **triggerName = &(createTriggerStmt->trigname);
+	AppendShardIdToName(triggerName, shardId);
+
+	char **relationSchemaName = &(relation->schemaname);
+	SetSchemaNameIfNotExist(relationSchemaName, schemaName);
```

Comment:
Not related to this PR but I think the name of this function and what it is doing doesn't match well.  To me the following seems more clear:

```c
if (relation->schemaname == NULL) {
       SetSchemaName( & relation->schemaname, schemaName);
}
```